### PR TITLE
Move to proposal-commit logic

### PR DESCRIPTION
**Pull request by kkohbrok**

### CODE REVIEW

File: draft-robert-mimi-delivery-service.md [hunk 1]
```diff
@@ -78,45 +78,49 @@ TODO: client identifiers definitions are preliminary.
...
+proposals include any proposals in {{!RFC9420}}:
 
-Follower servers can send requests the following:
+* Add proposals (to add new group members)
+* Remove proposals (to remove existing group members)
+* Update proposals (to update a group members's key material)
+* PSK proposals (to inject additional key material into the group)
+* Re-Init proposals (to re-initialize the group, e.g. with a new ciphersuite)
+* Group Context Extensions (to modify data in GroupContext Extensions, e.g. to
+  add/remove/update external senders)
```
Comment (raphaelrobert) [inline draft-robert-mimi-delivery-service.md:106]: Don't we want to allow any kind of proposal as long as it is supported by the group?

File: draft-robert-mimi-delivery-service.md [hunk 2]
```diff
@@ -78,45 +78,49 @@ TODO: client identifiers definitions are preliminary.
...
 
-Locally, the MIMI DS thus provides the following interface:
+Clients that are not group members can send an (external) commit to add
+themselves to a group.
 
-* Provide a Membership list (and associated metadata)
-* Process a query from a client and stage the corresponding changes
-* Process a query from a follower server and stage the corresponding changes
-* Merge the changes resulting from a previously processed query
+Group members can send an (external) commit to re-join the group (e.g. if they
```
Comment (raphaelrobert) [inline draft-robert-mimi-delivery-service.md:119]:
```suggestion
Group members can send an external commit to re-join the group (e.g. if they
```

File: draft-robert-mimi-delivery-service.md [hunk 3]
```diff
@@ -862,23 +772,22 @@ The DS responds with the KeyPackages of all clients listed in the request.
...
+
+The DS responds with the group's current GroupInfo.
 
 **Validation:**
 
-* The MLSGroupUpdate MUST contain a PublicMessage that contains a commit with a
-  re-init proposal.
-* The GroupID in the re-init proposal MUST point to another group on this hub,
-  which has a MIMI DS protocol version that is greater or equal than this group.
+* The DS SHOULD verify that the sender of the request is authorized to retrieve
```
Comment (raphaelrobert) [inline draft-robert-mimi-delivery-service.md:789]: Maybe MUST?

**Review by raphaelrobert — COMMENTED** [review]
Looks good in general, just some nits

File: draft-robert-mimi-delivery-service.md (hunk 1 above)
Comment (kkohbrok) [reply draft-robert-mimi-delivery-service.md:108]: Good point. I mostly wanted to give an overview over the operations that are generally possible. But of course, the MIMI DS protocol itself does not impose any particular restrictions.

**Pull request merged**